### PR TITLE
Revert to using a constructor for PerformanceObserver.

### PR DESCRIPTION
**Pull request by mpb**
As per #40, revert from window.performance.createPerformanceObserver to using new PerformanceObserver

### CODE REVIEW

**Comment by igrigorik** [comment]
Hmm, we'll have a conflict with https://github.com/w3c/performance-timeline/pull/37 -- that pull changes typeFilter to entryType to be consistent with getEntries*. But that's more of a merge concern.. 

Perhaps we should add a note indicating that the language should be updated once https://github.com/w3c/performance-timeline/issues/40#issuecomment-122975250 is resolved? 

/cc @bzbarsky @annevk

**Comment by igrigorik** [comment]
> I filed bugs against IDL and also HTML, but W3C Bugzilla is down at the moment.

@annevk do you have links to these?

**Comment by annevk** [comment]
See https://www.w3.org/Bugs/Public/show_bug.cgi?id=24652 and its dependencies.

**Comment by mpb** [comment]
Resolved the conflict with #37 
Although `entryType` sounds a little funny for a sequence name.
```WebIDL
dictionary PerformanceObserverInit {
    required sequence<DOMString> entryType;
};
```

File: index.html [hunk 1]
```diff
@@ -225,42 +225,35 @@
...
+            </dd>
+          </dl>
+
+          <p>The `observe(options)` method must run these steps:</p>
+          <ol>
+            <li>If _options'_ `entryType` attribute is not present, <a href="http://heycam.github.io/webidl/#dfn-throw">throw</a> a JavaScript `TypeError`.</li>
+            <li>Filter unsupported <a href="#widl-PerformanceEntry-entryType">entryType names</a> within the `entryType` sequence, and replace the `entryType` sequence with the new filtered sequence.</li>
+            <li>If the _options'_ `entryType` attribute is an empty sequence, <a href="http://heycam.github.io/webidl/#dfn-throw">throw</a> a JavaScript `TypeError`.</li>
+            <li>Add a new <a>registered performance observer</a> with the <a href="http://www.w3.org/TR/dom/#context-object">context object</a> as the <dfn>observer</dfn> and _options_ as the `options`.</li>
```
Comment (igrigorik) [inline index.html]: I'm a bit confused by this last line.. "context object as the observer"?

**Comment by toddreifsteck** [comment]
Per the 8/12 call, altering entryType to entryTypes likely makes sense given it is more than one.

**Comment by mpb** [comment]
PTAL

**Comment by igrigorik** [comment]
Looks good, thanks Michael!

Based on the discussion on the WG call today, I think we're all in agreement that the `new PerformanceObserver` interface is what we want.. And there has already been some confusion around this as current spec is showing the wrong pattern. With that in mind, I'm merging this to get the right interface in front of folks reading the spec, and I've opened a separate issue ( see https://github.com/w3c/performance-timeline/issues/44) to clarify the spec language.

**Pull request merged**